### ISSUE TITLE
NetVLAD effectiveness in Gazebo Environments

### ISSUE THREAD

**Issue by Quest2GM** [Relja/netvlad#45]
Hi @Relja,

NetVLAD is trained on real-world images, but I was wondering how effective your pre-trained models would work in simulation camera images (like in a Gazebo environment). To be more specific, my Gazebo world has simple obstacles like bookshelves, garbage cans and walls. A sample camera image taken from my robot in the simulation is shown here:

![tempimg2](https://user-images.githubusercontent.com/39883561/175492781-f854003e-6abb-441f-a6c5-f463afa91f87.jpg)

To your knowledge, has anyone tried this out? If so, what do you think are the best pre-trained models to try out? I'm trying to avoid training a custom model since building a dataset is super tedious in my instance.

Thanks for your response!
Quest2GM

**Comment by Relja** [comment]
Hi,
Hmm, sorry but I don't know of any work that uses synthetic images. It's hard to tell as real vs synthetic is quite different so I can only speculate really, but I wouldn't be surprised if it still worked decently enough. Probably best is VGG-16 + NetVLAD + whitening trained on Pitts 30k. If I was you I'd actually start with the simplest thing possible, e.g. take a resnet50 trained on ImageNet and do average or max pool the final activations, L2 normalize, and see how well that works. It might already be good enough for simple (if they are simple) synthetic scenes, and only later progress to NetVLAD if you want to squeeze more performance out. Though also it would be great to actually train the network - it's surprising that making the dataset is tedious since usually that's the apeal of synthetic data - easy generation of lots of it with good ground truth.
Best,
Relja

**Comment by Quest2GM** [comment]
Just wanted to inform that it does work pretty decently on the synthetic images, given that you tune the vector distance thresholds properly.

**Comment by Relja** [comment]
Thanks for the followup, I'm glad it worked out for you!
